projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -1026,6 +1049,73 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
         return new IgxGridRow(this, index, rec);
     }
 
+    /**
+     * @hidden @internal
+     */
+    public allRows(): RowType[] {
+        return this.dataView.map((rec, index) => this.createRow(index));
+    }
+
+    /**
+     * Returns the collection of `IgxGridRow`s for current page.
+     *
+     * @hidden @internal
+     */
+    public dataRows(): RowType[] {
+        return this.allRows().filter(row => row instanceof IgxGridRow);
+    }
+
+    /**
+     * Returns an array of the selected `IgxGridCell`s.
+     *
+     * @example
+     * ```typescript
+     * const selectedCells = this.grid.selectedCells;
+     * ```
+     */
+    public get selectedCells(): CellType[] {
+        return this.dataRows().map((row) => row.cells.filter((cell) => cell.selected))
+            .reduce((a, b) => a.concat(b), []);
+    }
+
+    /**
+     * Returns a `CellType` object that matches the conditions.
+     *
+     * @example
+     * ```typescript
+     * const myCell = this.grid1.getCellByColumn(2, "UnitPrice");
+     * ```
+     * @param rowIndex
+     * @param columnField
+     */
+    public getCellByColumn(rowIndex: number, columnField: string): CellType {
```

Comment:
Why don't we just inherit that method from the grid-base directive and enhance it's implementation under all other grid classes?

Reply:
good idea, can possibly bring back this methods in gridbase too. But inheriting such a short method would make it less readable I think.